sonic-net/sonic-buildimage
# Enhanced redis security: ACL and TLS

#### Why I did it
Enhance Redis Security.
Add Redis ACL support
Add Redis TLS support

| PR title | state | context |
| ---------------------------------------------------------------------------------------------------- | ------------------------------------------------------------------------------------------------------------------- | ------------------------------------------------------------------------------------------------------------------------ | 
| [[sonic-py-swsssdk] Added support for Redis](https://github.com/sonic-net/sonic-py-swsssdk/pull/144) | ![GitHub issue/pull request detail](https://img.shields.io/github/pulls/detail/state/sonic-net/sonic-py-swsssdk/144) | ![GitHub pull request check contexts](https://img.shields.io/github/status/contexts/pulls/sonic-net/sonic-py-swsssdk/144) |
| [[sonic-swss-common] Added support for Redis](https://github.com/sonic-net/sonic-swss-common/pull/844) | ![GitHub issue/pull request detail](https://img.shields.io/github/pulls/detail/state/sonic-net/sonic-swss-common/844) | ![GitHub pull request check contexts](https://img.shields.io/github/status/contexts/pulls/sonic-net/sonic-swss-common/844) |
| [[sonic-platform-common]Added support for Redis](https://github.com/sonic-net/sonic-platform-common/pull/424) | ![GitHub issue/pull request detail](https://img.shields.io/github/pulls/detail/state/sonic-net/sonic-platform-common/424) | ![GitHub pull request check contexts](https://img.shields.io/github/status/contexts/pulls/sonic-net/sonic-platform-common/424) |
| [[redis-dump-load]Added support for Redis](https://github.com/p/redis-dump-load/pull/67) | ![GitHub issue/pull request detail](https://img.shields.io/github/pulls/detail/state/p/redis-dump-load/67) | ![GitHub pull request check contexts](https://img.shields.io/github/status/contexts/pulls/p/redis-dump-load/67) |

##### Work item tracking
- Microsoft ADO **(number only)**:

#### How I did it
Add Redis ACL support by creating 2 users nad configure them in Redis Server.
one admin user who has access to any commands and operations like write/read
one monitor user who has access to read-only
Add Redis TLS by enable it in Redis Server.

#### How to verify it
Good flow:
each flow that exists today that uses the Redis DB should continue to be supported
Bad flow:
Add a new user which not belong to the admin group, and try to write to the Redis DB, this operation should be blocked.



#### Which release branch to backport (provide reason below if selected)



#### Tested branch (Please provide the tested image version)




#### Description for the changelog




#### Link to config_db schema for YANG module changes


#### A picture of a cute animal (not mandatory but encouraged)
